src/auth_db/mod.rs

```diff
@@ -0,0 +1,139 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::{
+  error::Error,
+  fmt::{self, Display, Formatter},
+};
+
+use hex;
+use reqwest::{Client, Error as RequestError, StatusCode, Url, UrlError};
+
+use settings::Settings;
+
+#[cfg(test)]
+mod test;
+
+#[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, PartialEq)]
+pub enum BounceType
+{
+  Hard = 1,
+  Soft = 2,
+  Complaint = 3,
+}
+
+#[derive(Clone, Copy, Debug, Deserialize)]
+pub struct BounceRecord
+{
+  #[serde(rename = "bounceType")]
+  pub bounce_type: BounceType,
+  #[serde(rename = "createdAt")]
+  pub created_at: u64,
+}
+
+#[derive(Debug)]
+pub struct DbError
+{
+  description: String,
+}
+
+impl DbError
+{
+  pub fn new(description: String) -> DbError
+  {
+    DbError { description }
+  }
+}
+
+impl Error for DbError
+{
+  fn description(&self) -> &str
+  {
+    &self.description
+  }
+}
+
+impl Display for DbError
+{
+  fn fmt(&self, formatter: &mut Formatter) -> fmt::Result
+  {
+    write!(formatter, "{}", self.description)
+  }
+}
+
+impl From<UrlError> for DbError
```

Comment:
This `From` trait was the crucial missing link in my understanding of how we can implement clean error handling in this repo. We can lean on it for #8.